hell/adventcalendar/2024/14/index.md

```diff
@@ -12,147 +12,185 @@ active: true
+    </li>
+
+    <li>
+      <span>...</span>
+      <span class="sr-only">skipping pages 10 to 12</span>
+    </li>
+
+    <li>
+      <a href="/9/">
+        <img src="next-page-icon.png" alt="Next Page, Page 9" />
+      </a>
+    </li>
+
+    <li>
+      <a href="/12/">
+        <span class="sr-only">Last Page, Page 12</span>
+        <svg role="presentation" focusable="false" xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 24 24">
+          <path d="M10 6l6 6-6 6 1.41 1.41L18.41 12 11.41 4.59 10 6z" />
+          <path d="M4 6l6 6-6 6 1.41 1.41L12.41 12 5.41 4.59 4 6z" />
+        </svg>
+      </a>
+    </li>
+  </ul>
 </nav>
 ```
 
-**Use of Images and Icons**:
-
-* To prevent symbols from being redundantly read aloud by screen readers, SVGs and images are given `aria-hidden="true"`. The link functionality is described exclusively via the aria-label in the `<a>` tag (e.g., "Next Page"), ensuring a clear and precise output for screen readers.
-<!-- MM: Why aria-label and not visually hidden text? -->
-<!-- KS: This is my primary question throughout this post. Text content
-     of `<a>` elements is very well understood as an accessible name,
-     and should be preferred over aria- attributes. -->
-* The alt attribute for images remains empty (`alt=""`) to prevent purely visual images from being unnecessarily read aloud. Since the link’s functionality is already described by the `aria-label`, this avoids redundant information for screen reader users.
-<!-- MM: But instead of the aria-label you could use the alt, no? Why is aria-label better? -->
-<!-- MM: Why the extra aria-hidden=true in the code example? -->
-* SVG icons also include the `focusable="false"` attribute to prevent them from being focusable in keyboard navigation. This attribute is unnecessary for images, as they are automatically excluded from keyboard focus.
-<!-- MM: Do you know if there's a browser or a situation where this is still the case? Afaik SVGs only use to be focusable in IE 11. -->
+**Using SVGs**
+- Icons should always have descriptive text to make their function clear for screenreader users. Instead of describing the SVG itself (e.g. “Double arrow pointing left”), the purpose should be communicated (e.g. “First page”).
+- Using a `<span>` element with the [screenreader-only class .sr-only](https://webaim.org/techniques/css/invisiblecontent/) is considered a best practice. This class hides the text visually for sighted users but makes it accessible to screenreaders. One advantage over using an `aria-label` on the `<a>` element is that the text is read by screenreaders with the correct language and pronunciation.
+	- Example: An English text like “First Page” is read correctly with an English voice when using the `.sr-only` class. However, an `aria-label` text might sometimes be read in the screenreader’s language (e.g. with a German voice and pronunciation), which can sound unnatural and difficult to understand.
```

Comment:
Can you point to a resource that explains that in detail? And do you have data about different screen reader / browser pairings? Those are quite interesting claims I haven't heard before and I'd love to see that backed by data.

Reply:
Thank you for your feedback! The behavior I described was based on my tests using NVDA (Version 2024.4) with Chrome (Version 130) and German as the language preference, where I observed that aria-label text in English was read with a German voice and pronunciation, whereas text provided via the .sr-only class was read with the correct English voice. This behavior has also been documented in [this GitHub issue](https://github.com/nvaccess/nvda/issues/16285). However, when testing with VoiceOver on macOS, I found no difference between the two methods. I haven’t yet tested this with JAWS or other screen reader-browser pairings.

I can include my test scenario to clarify that this behavior is not standard. How about the following phrasing?

Using a `<span>` element with the [screenreader-only class .sr-only](https://webaim.org/techniques/css/invisiblecontent/) can be a practical approach. This class hides the text visually for sighted users but makes it accessible to screenreaders. In certain cases, it may have an advantage over using an `aria-label` on the `<a>` element because screenreaders read the text with the correct language and pronunciation preferences.
- Tested Behavior: When using NVDA (Version 2024.4) with Chrome (Version 130) and German as the language preference, English text in an `aria-label` was read with a German voice and pronunciation. However, when the text was provided via the `.sr-only` class, it was read correctly with an English voice. This behavior also has been documented in [NVDA GitHub Issue #16285](https://github.com/nvaccess/nvda/issues/16285).
- With VoiceOver on macOS, no difference between the two methods was observed.
- Note: This has not been tested with JAWS or other screen readers and may vary depending on the browser and screen reader combination.

Reply:
Much much better, thank you!